web/src/app/pages/create-survey/create-survey.component.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ActivatedRoute} from '@angular/router';
+import {Component, OnInit} from '@angular/core';
+import {SurveyService} from 'app/services/survey/survey.service';
+import {FormGroup, FormBuilder} from '@angular/forms';
+import {NavigationService} from 'app/services/navigation/navigation.service';
+import {Survey} from 'app/models/survey.model';
+
+@Component({
+  selector: 'create-survey',
+  templateUrl: './create-survey.component.html',
+  styleUrls: ['./create-survey.component.scss'],
+})
+export class CreateSurveyComponent implements OnInit {
+  readonly titleControlKey = 'title';
+  readonly descriptionControlKey = 'description';
+  private currentSurveyId: string | null = null;
+  formGroup: FormGroup;
+
+  constructor(
+    private surveyService: SurveyService,
+    private navigationService: NavigationService,
+    route: ActivatedRoute
+  ) {
+    this.formGroup = new FormBuilder().group({
+      [this.titleControlKey]: '',
+      [this.descriptionControlKey]: '',
+    });
+    navigationService.init(route);
+  }
+
+  ngOnInit(): void {
+    this.navigationService.getSurveyId$().subscribe(surveyId => {
+      if (surveyId) {
+        this.surveyService.activateSurvey(surveyId);
```

Comment:
Is the survey first created as a side-effect of this call? If we create the survey only after this first step is completed, we can avoid having a bunch of abandoned surveys in the db with no name/description. That would also make creation explicit.